dataverse-webapp/src/main/webapp/WEB-INF/templates/dataset/inputRenderer/controlledVocabEnhancedInputField.xhtml

```diff
@@ -0,0 +1,135 @@
+<ui:composition xmlns="http://www.w3.org/1999/xhtml"
+                xmlns:h="http://java.sun.com/jsf/html"
+                xmlns:ui="http://java.sun.com/jsf/facelets"
+                xmlns:p="http://primefaces.org/ui"
+                xmlns:jsf="http://xmlns.jcp.org/jsf">
+
+    <!--@elvariable id="datasetField" type="edu.harvard.iq.dataverse.persistence.dataset.DatasetField"-->
+    <!--@elvariable id="inputRenderer" type="edu.harvard.iq.dataverse.dataset.metadata.inputRenderer.VocabSelectEnhancedInputFieldRenderer"-->
+
+    <div class="control-label-with-tooltip">
+        <p:outputLabel styleClass="control-label #{hiddenLabel ? 'sr-only' : ''}" for="#{datasetField.datasetFieldType.allowMultiples ? 'suggestionTextEnhancedMultipleSelectWidget' : 'suggestionTextEnhancedSingleSelectWidget'}">
+            <h:outputText value="#{datasetField.datasetFieldType.localeTitle}" escape="false" />
+            <h:outputText value=" #{labelFieldNumber}" rendered="#{not (labelFieldNumber eq 0)}" />
+            <h:outputText styleClass="glyphicon glyphicon-asterisk text-danger" value="" rendered="#{datasetField.datasetFieldType.requiredInDataverse}" />
+        </p:outputLabel>
+        <span class="glyphicon glyphicon-question-sign tooltip-icon"
+              jsf:rendered="#{(not empty datasetField.datasetFieldType.localeDescription) and !hiddenLabel}"
+              tabindex="0" role="button"
+              aria-label="#{bundle['common.forms.field.tooltip.ariaLabel']} #{datasetField.datasetFieldType.localeTitle}"
+              data-toggle="tooltip" data-placement="auto right"
+              data-original-title="#{datasetField.datasetFieldType.localeDescription}"></span>
+    </div>
+
+
+    <h:head>
+        <script>
+            // Helper function to find a scrollable element by traversing the DOM tree
+            function findScrollableElement(element) {
```

Comment:
I've tested it and it works. I would like to not move it to global JS as this behavior is strictly related to this input only. I think this placement makes a lot of sense.

Reply:
You're already defining those methods as global javascript functions. The script element will be included for every field of this type. The function names are also rather generic, if someone would add a method with the same name (eg `findScrollableElement`) as a script for another field, the actual method being called would depend on whatever field of those types was last included in the page.
Putting it in `dvjs` would allow to scope it inside `DvJS.ControlledVocabEnhancedInputField` and avoid those issues.

Reply:
Done